### PR TITLE
Add preliminary Ingres/Vector/Actian Avalanche support

### PR DESCRIPTION
**Pull request by clach04**
Ingres, like most DBMS implementation, DDL should contain the exact
length/precision and scale lengths. The defaults if omitted work
but are not likely to be desired.

### CODE REVIEW

**Comment by clach04** [comment]
Thanks for making this and https://github.com/wireservice/csvkit available. I just used csvkit for the first time this week and whilst I was already using a similar tool, csvkit is significantly more useful and robust 👍 

I opened this PR mostly to start discussion, although if it's approved that would be nice too ;-)

What do you think about reversing the logic for the string length code and also the decimal precision/scale code so that they always fire irrespective of the dialect? I.e. the lines with dialect checks:

 * https://github.com/wireservice/agate-sql/blob/master/agatesql/table.py#L197
 * https://github.com/wireservice/agate-sql/blob/master/agatesql/table.py#L209

Either remove the conditional check OR replace with an alternative check if there are some backends which should not apply this logic. Which backends should this NOT be done for? sqlite doesn't need it from a DDL perspective (as sqlite essentially ignores the type spec) but I'm not sure about the SQLAlchemy dialect. I've not tested this out, I've only tested the code in the PR hence the question and the conservative code change.

This works great when used with https://github.com/clach04/ingres_sa_dialect which is a (perpetually) in-progress Ingres dialect for SQLAlchemy.

File: agatesql/table.py [hunk 1]
```diff
@@ -194,19 +194,26 @@ def make_sql_table(table, table_name, dialect=None, db_schema=None, constraints=
...
         if constraints:
-            if isinstance(column.data_type, agate.Text) and dialect == 'mysql':
+            if isinstance(column.data_type, agate.Text) and dialect in ('ingres', 'mysql'):
                 length = table.aggregate(agate.MaxLength(column_name)) * decimal.Decimal(col_len_multiplier)
                 if length > 21844:
                     # @see https://dev.mysql.com/doc/refman/5.7/en/string-type-overview.html
                     sql_column_type = TEXT
+                elif dialect == 'ingres' and length > 16000:
+                    # @see https://docs.actian.com/actianx/11.2/index.html#page/SQLRef%2FCharacter_Data_Types.htm%23
+                    # Could be longer but ~16K is a reasonable cut off to switch to CLOB
```
Comment (jpmckinney) [inline agatesql/table.py:204]: The preceding condition is "length > 21844" (which is 65,535 bytes divided by 3). The Actian docs mention 32,000 bytes, so not sure how you chose 16,000. I'll go with 10,666 (32,000 divided by 3) https://docs.actian.com/ingres/11.2/index.html#page/SQLRef/Character_Data_Types.htm

**Comment by jpmckinney** [comment]
Added in 4f8ad30

File: agatesql/table.py (hunk 1 above)
Comment (clach04) [reply agatesql/table.py:204]: Agreed, ~10K is a better safe/sane default. It was a while ago so like you, I cannot figure out how I chose ~16K either 😜

Thanks for taking the time to review and improve the original idea! Happy New Year 🎉

**Pull request closed without merging**